### PR TITLE
[wake-lock] Add tests for query() method

### LINKED ISSUE #16285: [wake-lock] Add tests for `query()` method
See spec change at https://github.com/w3c/wake-lock/pull/172

### PR DESCRIPTION
**Pull request by Honry**
Fixed #16285

### CODE REVIEW

**Review by marcoscaceres — CHANGES_REQUESTED** [review]
Generally, the tests look good... but the messages are not very helpful when things go wrong... Id expect:

```suggestion
  assert_true(lock1.active, "lock1 must have been activated.");
```

And so on... I'd suggest just quickly going over the failure messages and making sure they make sense for when things fail (i.e., it's always clear why something failed).

**Comment by Honry** [comment]
@marcoscaceres, thank you for the suggestion, done. :)

File: wake-lock/wakelock-query.https.any.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  assert_true(lock2.active, "lock2 must have been activated");
+  assert_false(lock3.active, "lock3 must have been activated");
+  assert_false(lock4.active, "lock4 must have been activated");
+
+  // query activated screen WakeLock
+  const lockList1 = lock1.query({ type: "screen", active: true });
+  assert_equals(lockList1.size, 2, "Size of activated screen WakeLock instance list");
+  for (let lock of lockList1) {
+    assert_true(lock instanceof WakeLock, "query() must have returned WakeLock instance");
+    assert_equals(lock.type, "screen", "WakeLock type");
```
Comment (marcoscaceres) [inline wake-lock/wakelock-query.https.any.js]:
```suggestion
    assert_equals(lock.type, "screen", "Got wrong WakeLock type");
```

File: wake-lock/wakelock-query.https.any.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  }
+
+  // query activated system WakeLock
+  const lockList2 = lock1.query({ type: "system", active: true });
+  assert_equals(lockList2.size, 0, "Size of activated system WakeLock instance list");
```
Comment (marcoscaceres) [inline wake-lock/wakelock-query.https.any.js]:
```suggestion
  assert_equals(lockList2.size, 0, "Size of activated system WakeLock instance list is incorrect");
```

**Review by marcoscaceres — CHANGES_REQUESTED** [review]
Couple of others... but starting to look really good @Honry

File: wake-lock/wakelock-query.https.any.js (hunk 1 above)
Comment (Honry) [reply wake-lock/wakelock-query.https.any.js]: A bit concern here, the failure message from test run for `assert_equals` would look like:

_assert_equals: WakeLock type expected "screen" but got "xxx"_

That's why I only write "WakeLock type" as message.

Comment (marcoscaceres) [reply wake-lock/wakelock-query.https.any.js]: Ah! ok, good call.

File: wake-lock/wakelock-query.https.any.js (hunk 2 above)
Comment (marcoscaceres) [reply wake-lock/wakelock-query.https.any.js]: @Honry if you think this one is fine also, please ignore.

**Review by marcoscaceres — APPROVED** [review]
lgtm... however, @kenchris knows the spec text best, so he is in the best position to make sure all parts of the algorithm are tested.

**Comment by Honry** [comment]
Thank you @marcoscaceres! Now ping @kenchris.

File: wake-lock/wakelock-query.https.any.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    "",
+    null,
+    true,
+    123,
+    {},
+    Symbol(),
+    () => {},
+    self
+  ];
+  const lock = new WakeLock("screen");
```
Comment (kenchris) [inline wake-lock/wakelock-query.https.any.js]: what about no argument?

File: wake-lock/wakelock-query.https.any.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+}, "Filter constructed WakeLock instances with 'active' option");
+
+promise_test(async t => {
+  const lock1 = new WakeLock("screen");
+  const lock2 = new WakeLock("screen");
+  const lock3 = new WakeLock("system");
+  const lock4 = new WakeLock("system");
+
+  const controller = new AbortController();
+  await lock1.request({ signal: controller.signal });
```
Comment (kenchris) [inline wake-lock/wakelock-query.https.any.js]: Notice that we agreed on removing abortcontroller and there is a WIP PR for it

Comment (Honry) [reply wake-lock/wakelock-query.https.any.js]: Yeah, I knew the change, and will replace all `AbortController` with `release()` once your PR is landed.

File: wake-lock/wakelock-query.https.any.js (hunk 3 above)
Comment (Honry) [reply wake-lock/wakelock-query.https.any.js]: Uh, sorry, I didn't get the point. Isn't the "WakeLockType" a required argument for the constructor?

Comment (kenchris) [reply wake-lock/wakelock-query.https.any.js]: but you want to check that it fails no? :)

Comment (Honry) [reply wake-lock/wakelock-query.https.any.js]: That'd been tested in this file: https://github.com/web-platform-tests/wpt/blob/master/wake-lock/wakelock-type.https.html

And here, I intended to test the invalid type argument for `query()` method.

**Comment by Honry** [comment]
Close it since `query` method has been removed in https://github.com/w3c/wake-lock/pull/200/.

**Pull request closed without merging**